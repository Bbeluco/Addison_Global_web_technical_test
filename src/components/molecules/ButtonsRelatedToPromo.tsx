import React from 'react';
import {StyleSheet, View} from 'react-native';
import {CustomButton} from '../atoms/CustomButton';
import {stylesType} from '../../types/stylesType';

export function ButtonsRelatedToPromo({
  styleParent,
}: Omit<stylesType, 'styleChildren'>): React.JSX.Element {
  return (
    <View style={styleParent}>
      <CustomButton
        text={'Terms & Conditions'}
        styleParent={[
          stylesTermsAndConditions.button,
          stylesCustomButtonDefault.button,
        ]}
        styleChildren={[
          stylesTermsAndConditions.text,
          stylesCustomButtonDefault.text,
        ]}
      />

      <CustomButton
        text={'Join Now'}
        styleParent={[stylesJoinNow.button, stylesCustomButtonDefault.button]}
        styleChildren={[stylesJoinNow.text, stylesCustomButtonDefault.text]}
      />
    </View>
  );
}

const stylesCustomButtonDefault = StyleSheet.create({
  button: {
    width: '40%',
    borderRadius: 5,
    alignItems: 'center',
  },
  text: {
    padding: 5,
    fontWeight: 'bold',
  },
});

const stylesTermsAndConditions = StyleSheet.create({
  button: {
    borderWidth: 2,
    borderColor: 'gray',
  },
  text: {
    color: 'gray',
  },
});

const stylesJoinNow = StyleSheet.create({
  button: {
    backgroundColor: 'gray',
  },
  text: {
    color: 'white',
  },
});
