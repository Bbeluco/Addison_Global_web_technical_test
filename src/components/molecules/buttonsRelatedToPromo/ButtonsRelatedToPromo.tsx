import React from 'react';
import {View} from 'react-native';
import {CustomButton} from '@components/atoms/customButton/CustomButton';
import {ButtonsRelatedToPromoType} from '@customTypes/buttonsRelatedToPromo';
import {
  stylesCustomButtonDefault,
  stylesTermsAndConditions,
  stylesJoinNow,
} from './styles';

export function ButtonsRelatedToPromo({
  styleParent,
  textJoinNowButtonText,
  textTermsAndConditionsButtonText,
}: ButtonsRelatedToPromoType): React.JSX.Element {
  return (
    <View style={styleParent}>
      <CustomButton
        text={textTermsAndConditionsButtonText}
        styleParent={[
          stylesTermsAndConditions.button,
          stylesCustomButtonDefault.button,
        ]}
        styleChildren={[
          stylesTermsAndConditions.text,
          stylesCustomButtonDefault.text,
        ]}
      />

      <CustomButton
        text={textJoinNowButtonText}
        styleParent={[stylesJoinNow.button, stylesCustomButtonDefault.button]}
        styleChildren={[stylesJoinNow.text, stylesCustomButtonDefault.text]}
      />
    </View>
  );
}
