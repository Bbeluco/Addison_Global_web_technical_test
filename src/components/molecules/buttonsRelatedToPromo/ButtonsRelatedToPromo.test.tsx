import React from 'react';
import {render, screen} from '@testing-library/react-native';
import {ButtonsRelatedToPromo} from './ButtonsRelatedToPromo';

describe('<ButtonsRelatedToPromo />', () => {
  it('UI test', () => {
    const textJoinNowButtonText = 'Join Now';
    const textTermsAndConditionsButtonText = 'Terms & Conditions';
    render(
      <ButtonsRelatedToPromo
        styleParent={{}}
        textJoinNowButtonText={textJoinNowButtonText}
        textTermsAndConditionsButtonText={textTermsAndConditionsButtonText}
      />,
    );

    expect(screen.queryByText(textJoinNowButtonText)).not.toBeNull();
    expect(screen.queryByText(textTermsAndConditionsButtonText)).not.toBeNull();
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
