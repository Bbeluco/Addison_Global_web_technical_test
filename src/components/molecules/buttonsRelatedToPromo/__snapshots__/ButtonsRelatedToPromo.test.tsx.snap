// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonsRelatedToPromo /> UI test 1`] = `
<View
  style={{}}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "gray",
        "borderRadius": 5,
        "borderWidth": 2,
        "justifyContent": "center",
        "opacity": 1,
        "width": "40%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "gray",
          },
          {
            "fontSize": 12,
            "fontWeight": "bold",
            "padding": 5,
          },
        ]
      }
    >
      Terms & Conditions
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "gray",
        "borderRadius": 5,
        "justifyContent": "center",
        "opacity": 1,
        "width": "40%",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "white",
          },
          {
            "fontSize": 12,
            "fontWeight": "bold",
            "padding": 5,
          },
        ]
      }
    >
      Join Now
    </Text>
  </View>
</View>
`;
